modules/mono/csharp_script.cpp

```diff
@@ -975,12 +975,28 @@ void CSharpLanguage::reload_assemblies(bool p_soft_reload) {
 #endif
 
 		if (!scr->get_path().is_empty()) {
-			scr->reload(p_soft_reload);
+			// Constructing default properties while reloading the assembly can lead to their scripts being created and registered by this point
+			// Check if a script was created that way and let it take over the path rather than trying to reload this script which would fail
+			Ref<CSharpScript> existing_script;
+			bool script_already_exists = false;
+			String script_path = scr->get_path();
+
+			if (GDMonoCache::godot_api_cache_updated) {
+				script_already_exists = GDMonoCache::managed_callbacks.ScriptManagerBridge_GetExistingScriptBridgeForPath(&script_path, &existing_script);
+			}
+			if (script_already_exists) {
```

Comment:
This can be inlined to a single if statement.
```suggestion
			String script_path = scr->get_path();

			if (GDMonoCache::godot_api_cache_updated && GDMonoCache::managed_callbacks.ScriptManagerBridge_GetExistingScriptBridgeForPath(&script_path, &existing_script)) {
```